src/IndexerInterface.php

```diff
@@ -17,46 +17,81 @@
  * Copyright (c) 2018 (original work) Open Assessment Technologies SA;
  */
 
+declare(strict_types=1);
+
 namespace oat\tao\elasticsearch;
 
 use Iterator;
+use oat\tao\model\search\index\IndexDocument;
+use oat\tao\model\TaoOntology;
+use oat\taoResultServer\models\classes\ResultService;
 
 /**
  * Interface IndexerInterface
  */
 interface IndexerInterface
 {
-    /**
-     * Returns name of current using index
-     * @return mixed
-     */
-    public function getIndex();
+    public const ITEMS_INDEX = 'items';
+    public const TESTS_INDEX = 'tests';
+    public const TEST_TAKERS_INDEX = 'test-takers';
+    public const DELIVERIES_INDEX = 'deliveries';
+    public const DELIVERY_RESULTS_INDEX = 'delivery-results';
+    public const GROUPS_INDEX = 'groups';
+    public const UNCLASSIFIEDS_DOCUMENTS_INDEX = 'unclassifieds';
+    public const ASSETS_INDEX = 'assets';
+
+    public const MEDIA_CLASS_URI = 'http://www.tao.lu/Ontologies/TAOMedia.rdf#Media';
+
+    public const AVAILABLE_INDEXES = [
+        ResultService::DELIVERY_RESULT_CLASS_URI => self::DELIVERY_RESULTS_INDEX,
```

Comment:
maybe we can have this from a config injected or from the .env because we should not redeploy the code when we want to suppress or add a new index , 
also i did not see any parser on the elastic side (grok)